doc/api/deprecations.md

```diff
@@ -2651,6 +2651,39 @@ written twice. This introduces a race condition between threads, and is a
 potential security vulnerability. There is no safe, cross-platform alternative
 API.
 
+<a id="DEP0XXX"></a>
+### DEP0XXX: `module.parent`
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/32217
+    description: Documentation-only deprecation.
+-->
+
+Type: Documentation-only (supports [`--pending-deprecation`][])
+
+A CommonJS module can access the first module that required it using
+`module.parent`. This feature is deprecated because it does not work
+consistently in the presence of ECMAScript modules and because it gives an
+inaccurate representation of the CommonJS module graph.
+
+Some modules use it to check if they are the entry point of the current process.
+Instead, it is recommended to compare `require.main` and `module`:
+
+```js
+if (require.main === module) {
+  // Code section that will run only if current file is the entry point.
+}
+```
+
+When looking for the CJS modules that have required the current one,
```

Comment:
If I may ask: how do you do that from an ECMAScript module?

Reply:
I don't think their is an API available from a module itself; however if you are looking for a way to show the dependency graph, I think it can be achieved using [loader hooks](https://nodejs.org/docs/latest-v13.x/api/esm.html#esm_hooks).

Reply:
Surely not the best approach, but I found that by creating an error and inspecting the stack trace I could find out what modules imported and called the current one.